src/shared/conf-parser.c

```diff
@@ -1359,5 +1360,56 @@ int config_parse_hwaddrs(
         }
 }
 
+int config_parse_in_addr_non_null(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        struct in_addr *ipv4 = data;
+        struct in6_addr *ipv6 = data;
```

Comment:
So... the `data` pointer is to either of the two types, and not to a union. I guess this is OK, but please add a comment here. Normally one would expect `data` to be `union in_addr_union*` too.